src/main/java/org/assertj/core/api/AbstractFileAssert.java

```diff
@@ -338,6 +338,31 @@ public SELF hasBinaryContent(byte[] expected) {
     return myself;
   }
 
+  /**
+   * Verifies that the size of the actual {@code File} is <b>exactly</b> equal to the given size in bytes.
+   * <p>
+   * Example:
+   * <pre><code class='java'> File file = File.createTempFile(&quot;tmp&quot;, &quot;bin&quot;);
+   * Files.write(file.toPath(), new byte[] {1, 1});
+   *
+   * // assertion will pass
+   * assertThat(file).hasSize(2);
+   *
+   * // assertions will fail
+   * assertThat(file).hasSize(0);
+   * assertThat(file).hasSize(1);</code></pre>
+   *
+   * @param expected the expected {@code File}'s size in bytes.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code File} is {@code null}.
+   * @throws AssertionError if the actual {@code File} is not an existing file.
+   * @throws AssertionError if the size of the actual {@code File} is not equal to the given size.
+   */
+  public SELF hasSize(long expected) {
```

Comment:
Rename to `expected` to `expectedInBytes`, I wonder if we should rename the assertion `hasSizeInBytes` thoughts ?

Reply:
Makes sense, will refactor.